backend/courses/views.py

```diff
@@ -376,3 +385,147 @@ def get_queryset(self):
+    we either delete the friendship request if the sender hits the route, or we reject the request if the recipient hits this route.
+    """
+    model = Friendship
+    http_method_names = ["get", "post", "delete"]
+    permission_classes = [IsAuthenticated]
+    queryset = Friendship.objects.none()
+    serializer_class = FrienshipSerializer
+
+    # Is this PcxAutoSchema correct given the responses in each view function below?
+    schema = PcxAutoSchema(
+        response_codes={
+            reverse_func("friendships"): {
+                "GET": {
+                    200: "Friendships retrieved successfully.",
+                },
+                "POST": {
+                    200: "Friendship request handled successfully.",
+                    409: "Friendship request already exists",
+                    400: "Bad request.",
+                },
+                "DELETE": {
+                    200: "Friendship deleted successfully.",
+                    404: "Friendship does not exist.",
+                    409: "Friendship request already rejected.",
+                }
+            }
+        },
+        custom_parameters={
+            reverse_func("friendships"): {
+                "POST": [
+                    {
+                        "name": "friend_id",
+                        "in": "query",
+                        "description": "The ID of the user you are sending a friend request to or handling a request from.",  # noqa E501
+                        "schema": {"type": "int"},
+                        "required": True,
+                    },
+                ],
+                "DELETE": [
+                    {
+                        "name": "friend_id",
+                        "in": "query",
+                        "description": "The ID of the user you are ending/rejecting your friendship/friend request with.",  # noqa E501
+                        "schema": {"type": "int"},
+                        "required": True,
+                    },
+                ]
+            },
+        },
+        override_request_schema={
+            reverse_func("friendships"): {
+                "POST": {
+                    "type": "object",
+                }
+            }
+        }
+    )
+
+    def get_queryset(self):
+        return Friendship.objects.filter(
+            Q(sender=self.request.user) | Q(recipient=self.request.user),
+            status__ne=Friendship.Status.REJECTED
+        )
+                
+    def post(self, request):
+        res = {}
+        sender = request.user
+        recipient = get_object_or_404(User, id=request.GET.get("friend_id"))
+
+        existing_friendship = self.queryset.filter(
+            Q(recipient=recipient) | Q(sender=recipient)
+        ).first()
+
+        if not existing_friendship:
+            friendship = Friendship(sender=sender, recipient=recipient, status=Friendship.Status.SENT)
+            friendship.save()
+            res = model_to_dict(friendship)
```

Comment:
Should use `FriendshipSerializer` rather than `model_to_dict` (this is typically true, for security reasons).